src/pudl/transform/ferc714.py

```diff
@@ -315,35 +361,223 @@ def _post_process(df: pd.DataFrame, table_name: str) -> pd.DataFrame:
+    """
+
+    @classmethod
+    def run(
+        cls, raw_csv: pd.DataFrame, raw_xbrl_duration: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Build the table for the :ref:`core_ferc714__respondent_id` asset.
+
+        Process and combine the CSV and XBRL based data.
+
+        There are two main threads of transforms happening here:
+
+        * Table compatibility: The CSV raw table is static (does not even report years)
+          while the xbrl table is reported annually. A lot of the downstream analysis
+          expects this table to be static. So the first step was to check whether or not
+          the columns that we have in the CSV years had consistent data over the few XBRL
+          years that we have. There are a small number of eia_code's we needed to clean
+          up, but besides that it was static. We then convert the XBRL data into a static
+          table, then we concat-ed the tables and checked the static-ness again via
+          :meth:`ensure_eia_code_uniqueness`.
+        * eia_code cleaning: Clean up FERC-714 respondent names and manually assign EIA
+          utility IDs to a few FERC Form 714 respondents that report planning area demand,
+          but which don't have their corresponding EIA utility IDs provided by FERC for
+          some reason (including PacifiCorp). Done all via :meth:`spot_fix_eia_codes` &
+          EIA_CODE_FIXES.
+
+        """
+        table_name = "core_ferc714__respondent_id"
+        # CSV STUFF
+        csv = (
+            _pre_process_csv(raw_csv, table_name)
+            .pipe(_assign_respondent_id_ferc714, source="csv")
+            .astype({"eia_code": pd.Int64Dtype()})
+            .pipe(cls.spot_fix_eia_codes, "csv")
+            .pipe(cls.ensure_eia_code_uniqueness, "csv")
+            .assign(source="csv")
+        )
+        # XBRL STUFF
+        xbrl = (
+            rename_columns(
+                raw_xbrl_duration,
+                params=RenameColumns(columns=RENAME_COLS[table_name]["xbrl"]),
+            )
+            .pipe(_assign_respondent_id_ferc714, source="xbrl")
+            .pipe(cls.clean_eia_codes_xbrl)
+            .astype({"eia_code": pd.Int64Dtype()})
+            .pipe(cls.spot_fix_eia_codes, "xbrl")
+            .pipe(cls.ensure_eia_code_uniqueness, "xbrl")
+            .pipe(cls.convert_into_static_table_xbrl)
+            .assign(source="xbrl")
+        )
+        # CONCATED STUFF
+        df = (
+            pd.concat([csv, xbrl])
+            .reset_index(drop=True)
+            .convert_dtypes()
+            .pipe(cls.spot_fix_eia_codes, "combined")
+            .pipe(cls.ensure_eia_code_uniqueness, "combined")
+            .pipe(cls.condense_into_one_source_table)
+            .pipe(_fillna_respondent_id_ferc714_source, "csv")
+            # the xbrl version of this is fillna is not *strictly necessary*
+            # bc we are sorting the records to grab the xbrl record
+            .pipe(_fillna_respondent_id_ferc714_source, "xbrl")
```

Comment:
How come this is down in the combined section if it is specific to csv and xbrl?

Reply:
this is fair to flag bc i guess it does look weird. but i think this answer will be more clear if you read the response to your question above. but if its still unclear lmk! also i'll try to update the docs!

Reply:
That's kind of what I though--I think your docs updates will help!

Reply:
okay i added much more into that doc string. i'm going to resolve this comment but feel free to un-resolve it if it still needs attention.